### PR TITLE
fix(ground_filter): remove base_frame and fix ray_ground_filter 

### PR DESCRIPTION
**Pull request by badai-nguyen**
Signed-off-by: badai-nguyen <dai.nguyen@tier4.jp>

## Description
To solve the issue https://github.com/autowarefoundation/autoware.universe/issues/1384# , the base_frame changeable parameter will be removed as it is unnecessary and ray_ground_filter, scan_ground_filter is designed on fixed base_link

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] Confirm unchanged result before and after removing base_frame

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/ground_segmentation/src/ray_ground_filter_nodelet.cpp [hunk 1]
```diff
@@ -202,6 +169,7 @@ void RayGroundFilterComponent::ClassifyPointCloud(
       double local_max_slope = local_max_slope_;
       if (j == 0) {
         local_max_slope = initial_max_slope_;
+        prev_height = in_radial_ordered_clouds[i][j].point.z;
```
Comment (yukkysaito) [inline perception/ground_segmentation/src/ray_ground_filter_nodelet.cpp:172]: Is it unnecessary if input_frame is base link?

Comment (badai-nguyen) [reply perception/ground_segmentation/src/ray_ground_filter_nodelet.cpp:172]: thanks, it is unnecessary. I will remove it.

File: perception/ground_segmentation/src/ray_ground_filter_nodelet.cpp [hunk 2]
```diff
@@ -241,8 +208,8 @@ void RayGroundFilterComponent::ClassifyPointCloud(
           // if previous points wasn't ground
           if (!prev_ground) {
             if (
-              current_height <= general_height_threshold &&
-              current_height >= -general_height_threshold) {
+              current_height - prev_height <= general_height_threshold &&
+              current_height - prev_height >= -general_height_threshold) {
```
Comment (yukkysaito) [inline perception/ground_segmentation/src/ray_ground_filter_nodelet.cpp:212]: no need?

File: perception/ground_segmentation/src/ray_ground_filter_nodelet.cpp [hunk 3]
```diff
@@ -254,8 +221,8 @@ void RayGroundFilterComponent::ClassifyPointCloud(
           // check if previous point is too far from previous one, if so classify again
           if (
             points_distance > reclass_distance_threshold_ &&
-            (current_height <= general_height_threshold &&
-             current_height >= -general_height_threshold)) {
+            (current_height - prev_height <= general_height_threshold &&
+             current_height - prev_height >= -general_height_threshold)) {
```
Comment (yukkysaito) [inline perception/ground_segmentation/src/ray_ground_filter_nodelet.cpp:225]: same comment
https://github.com/autowarefoundation/autoware.universe/pull/1614/files#r949761213

**Review by yukkysaito — APPROVED** [review]
LGTM
@storrrrrrrrm @miursh 
is it ok?

File: perception/ground_segmentation/docs/scan-ground-filter.md [hunk 4]
```diff
@@ -30,7 +30,6 @@ This implementation inherits `pointcloud_preprocessor::Filter` class, please ref
 
 | Name                              | Type   | Default Value | Description                                                                   |
 | --------------------------------- | ------ | ------------- | ----------------------------------------------------------------------------- |
-| `base_frame`                      | string | "base_link"   | base_link frame                                                               |
```
Comment (miursh) [inline perception/ground_segmentation/docs/scan-ground-filter.md:33]: @badai-nguyen Could you add "input_frame" and "output_frame" as params? Also, I want you to write the limitation that  "input_frame" must be fixed to base_link.

File: perception/ground_segmentation/docs/ray-ground-filter.md [hunk 5]
```diff
@@ -26,7 +26,6 @@ This implementation inherits `pointcloud_preprocessor::Filter` class, please ref
 
 | Name                          | Type   | Description                                                                                                                                                                                                                    |
 | ----------------------------- | ------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
-| `base_frame`                  | string | The target frame of input points                                                                                                                                                                                               |
```
Comment (miursh) [inline perception/ground_segmentation/docs/ray-ground-filter.md:29]: @badai-nguyen Could you add "input_frame" and "output_frame" as params? Also, I want you to write the limitation that  "input_frame" must be fixed to base_link.

File: launch/tier4_perception_launch/launch/obstacle_segmentation/ground_segmentation/ground_segmentation.launch.py [hunk 6]
```diff
@@ -238,7 +238,6 @@ def create_common_pipeline(self, input_topic, output_topic):
                 parameters=[
                     self.ground_segmentation_param["common_ground_filter"]["parameters"],
                     self.vehicle_info,
-                    {"input_frame": "base_link"},
```
Comment (miursh) [inline launch/tier4_perception_launch/launch/obstacle_segmentation/ground_segmentation/ground_segmentation.launch.py]: Could you keep this line?

**Review by miursh — APPROVED** [review]
LGTM

**Pull request merged**